src/abstractdataframe/io.jl

```diff
@@ -1,17 +1,90 @@
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, false),
+         allcols::Bool = !get(io, :limit, false),
+         splitcols = get(io, :limit, false),
+         rowlabel::Symbol = :Row,
+         summary::Bool=true)
+
+Render a data frame to an I/O stream in MIME type `mime`.
+
+# Positional arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supported MIME types are: `"text/plain"`, `"text/html"`, `"text/latex"`,
+  `"text/csv"`, `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+
+# Keyword arguments
+- MIME type `"text/plain"` accepts all listed keyword arguments and therir behavior
+  is identical as for `show(::IO, ::AbstractDataFrame)`
+- MIME types `"text/html"`, `"text/latex"` accept only `allrows` and `allcols`
+  keyword arguments. They allow the user to decide to print all rows/columns
+  instead of the default values given in `HTML_NROWS`, `LATEX_NROWS`,
+  `HTML_NCOLS`, and `LATEX_NCOLS` constants.
+  Additionally `"text/html"` MIME type accepts `summary` keyword argument which
+  allows to choose whether to print a brief string summary of the data frame.
+- MIME types `"text/csv"`, `"text/tab-separated-values"` do not support any
+  keyword arguments.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
+Base.show(io::IO, mime::MIME, df::AbstractDataFrame;
+          allrows::Bool = !get(io, :limit, false),
+          allcols::Bool = !get(io, :limit, false),
+          summary::Bool=true) =
+    if mime isa MIME"text/html"
+        _show(io, mime, df, allrows=allrows, allcols=allcols, summary=summary)
+    elseif mime isa MIME"text/latex"
+        _show(io, mime, df, allrows=allrows, allcols=allcols)
+    else
+        throw(ArgumentError("Unknown MIME type: $mime"))
+    end
```

Comment:
Why are you even defining `Base.show(io::IO, mime::MIME, df::AbstractDataFrame)`?   Since the only two supported types are text/html and text/latex, and you do different things for these, you should just define those two, no?
```jl
"""
docstring
"""
```
function Base.show(io::IO, mime::MIME, df::AbstractDataFrame); end


Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame;
                    allrows::Bool = !get(io, :limit, false), allcols::Bool = !get(io, :limit, false), summary::Bool=true)
   = _show(io, mime, df, allrows=allrows, allcols=allcols, , summary=summary)
Base.show(io::IO, mime::MIME"text/latex", df::AbstractDataFrame;
                    allrows::Bool = !get(io, :limit, false), allcols::Bool = !get(io, :limit, false))
   = _show(io, mime, df, allrows=allrows, allcols=allcols)
```

Reply:
@nalimilan - how do you see the change now (when actually most of the cases have to be specialized anyway)? I can revert to the old design with all specialized methods (then the question is to what should we attach the docstring).

Reply:
The solution proposed by @stevengj looks good to attach the docstring to the right method, but AFAICT it has the drawback that we define a function which doesn't do anything. Maybe have it throw an "unsupported MIME type" error?

Reply:
OK